src/binding/petsc4py/src/petsc4py/PETSc/Mat.pyx

```diff
@@ -1514,7 +1935,24 @@ cdef class Mat(Object):
         CHKERR( MatLoad(self.mat, viewer.vwr) )
         return self
 
-    def convert(self, mat_type=None, Mat out=None):
+    def convert(self, mat_type: Type | str = None, Mat out=None) -> Mat:
+        """Convert the matrix type.
+
+        Collective.
+
+        Parameters
+        ----------
+        mat_type
+          The type of the new matrix. If `None` uses `Type.SAME`.
+        out
+           Optional return matrix. If `None`, inplace conversion is performed.
+           Otherwise, the matrix is reused.
```

Comment:
```suggestion
            The type of the new matrix. If `None` uses `Type.SAME`.
        out
            Optional return matrix. If `None`, inplace conversion is performed.
            Otherwise, the matrix is reused.
```